packages/core/src/factory.ts

```diff
@@ -3,165 +3,139 @@ import * as Either from 'fp-ts/lib/Either';
-                    components.mocker.mock(
-                      {
-                        resource,
-                        input: {
-                          validations: {
-                            input: inputValidationResult,
-                          },
-                          data: input,
-                        },
-                        config: configObj,
-                      },
-                      defaultComponents.mocker,
-                    )(components.logger.child({ name: 'NEGOTIATOR' })),
-                  ),
-                  TaskEither.map(output => ({ output, resource })),
-                );
-              } else if (components.forwarder) {
-                // forward request and set output from response
-                return pipe(
-                  components.forwarder.fforward(
-                    {
-                      resource,
-                      input: {
-                        validations: {
-                          input: inputValidationResult,
-                        },
-                        data: input,
-                      },
-                      config: configObj,
-                    },
-                    defaultComponents.forwarder,
-                  ),
-                  TaskEither.map(output => ({ output, resource })),
-                );
-              }
+            const inputValidationResult = inputValidations.concat(
+              pipe(
+                validateSecurity(input, resource),
+                fold<IPrismDiagnostic, IPrismDiagnostic[]>(() => [], value => [value]),
+              ),
+            );
 
-              return TaskEither.left(new Error('Nor forwarder nor mocker has been selected. Something is wrong'));
-            }),
-            TaskEither.map(({ output, resource }) => {
-              let outputValidations: IPrismDiagnostic[] = [];
-              if (resource && components.validator && components.validator.validateOutput) {
-                outputValidations = components.validator.validateOutput(
-                  {
+            if (resource && components.mocker && ((configObj as unknown) as IPrismConfig).mock) {
+              // generate the response
+              return pipe(
+                TaskEither.fromEither(
+                  components.mocker.mock({
                     resource,
-                    output,
+                    input: {
+                      validations: {
+                        input: inputValidationResult,
+                      },
+                      data: input,
+                    },
                     config: configObj,
+                  })(components.logger.child({ name: 'NEGOTIATOR' })),
+                ),
+                TaskEither.map(output => ({ output, resource })),
+              );
+            } else if (components.forwarder) {
+              // forward request and set output from response
+              return pipe(
+                components.forwarder.fforward({
```

Comment:
Is this forwarder logic used somewhere? We’re meant to be removing the forwarder as it exists in favour of clean and obvious separation of HTTP server and HTTP client. Anything that’s forwarder related which is not directly being used to power the HTTP Client should be removed and/or refactored as we go. Doesn’t have to be done in this PR though.

Reply:
Yes, I'm actually super ok removing the whole part and it should be really easy.

Reply:
Done in c4cfd24fcaac0e58a92bda907d519e5c1bf77064